colorama/ansitowin32.py

```diff
@@ -21,6 +21,35 @@ def is_a_tty(stream):
     return hasattr(stream, 'isatty') and stream.isatty()
 
 
+def check_windows_ansi_support():
+    try:
+        import winreg
+    except ImportError:
+        try:
+            import _winreg  # on Python 2
+        except ImportError:
+            return False
+        else:
+            winreg = _winreg
+
+    # Check the registry for release ID
+    key = r"SOFTWARE\Microsoft\Windows NT\CurrentVersion"
+    val = r"ReleaseID"
+    key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, key)
+    releaseId = int(winreg.QueryValueEx(key,val)[0])
+    winreg.CloseKey(key)
+
+    # Windows 10 supports ANSI cmd since release 1511
+    if releaseId >= 1511:
+        # but only on powershell
+        import psutil
```

Comment:
I think this isn't really related to powershell but to the `ENABLE_VIRTUAL_TERMINAL_PROCESSING` flag. colorama should enable this flag when supported and restore it's previous value on exit.

I think we can simply try to `SetConsoleMode(h, GetConsoleMode(h) | ENABLE_VIRTUAL_TERMINAL_PROCESSING)` to check for support of this (Need to be sure the stream is a tty/console before calling `{Get,Set}ConsoleMode`). I think it fails with `INVALID_ARGUMENT` when unsupported. But this needs some testing.

See: https://github.com/tartley/colorama/pull/133#issuecomment-315325623 & https://msdn.microsoft.com/en-us/library/windows/desktop/ms686033(v=vs.85).aspx